### DISCUSSION TITLE
Starting and Ending GCODE for Prusa Slicer?

### DISCUSSION THREAD

**Discussion by RB211** [hillsoftware/sv06#9] [General]
Has anyone created start and end GCODE for Prusa Slicer and profiles for this printer?

**Comment by hillsoftware** [comment]
Most people are just using the Prusa MK3S+ and changing the retraction to .5

I use the following for start g-code in prusa slicer for Bi-Linear version (would work with manual mesh too)

```G90 ; use absolute coordinates
M83 ; extruder relative mode
G28 ; home all
M420 S1
M104 S[first_layer_temperature] ; set extruder temp
M140 S[first_layer_bed_temperature] ; set bed temp
M190 S[first_layer_bed_temperature] ; wait for bed temp
M109 S[first_layer_temperature] ; wait for extruder temp
M117 Printing Purge Line
G1 Z2 F240
G1 X2 Y10 F3000
G1 Z0.28 F240
G92 E0
G1 Y190 E15 F1500 ; intro line
G1 X2.3 F5000
G92 E0
G1 Y10 E15 F1200 ; intro line
G92 E0
M117 Layer 1 of {total_layer_count}
```

For UBL I would use the following (basically same as above but M420 removed

```G90 ; use absolute coordinates
M83 ; extruder relative mode
G28 ; home all
G29 A ; Unified bed-level (BL-Touch)
G29 L0 ; Load Mesh Slot 0
G29 J3 ; Probe 9 points to check mesh
M104 S[first_layer_temperature] ; set extruder temp
M140 S[first_layer_bed_temperature] ; set bed temp
M190 S[first_layer_bed_temperature] ; wait for bed temp
M109 S[first_layer_temperature] ; wait for extruder temp
M117 Printing Purge Line
G1 Z2 F240
G1 X2 Y10 F3000
G1 Z0.28 F240
G92 E0
G1 Y190 E15 F1500 ; intro line
G1 X2.3 F5000
G92 E0
G1 Y10 E15 F1200 ; intro line
G92 E0
M117 Layer 1 of {total_layer_count}
```

I use the following for end g-code

```
G1 E-4 F2400 ; Retract 4mm and raise Z up
{if max_layer_z < max_print_height}G1 Z{z_offset+min(max_layer_z+2, max_print_height)} F600{endif} ; Move print head up
G1 X5 Y170 F{travel_speed*60} ; present print
{if max_layer_z < max_print_height-10}G1 Z{z_offset+min(max_layer_z+70, max_print_height-10)} F600{endif} ; Move print head further up
M140 S0 ; turn off heatbed
M104 S0 ; turn off temperature
M107 ; turn off fan
M84 X Y E ; disable motors
```

**Reply by RB211** [reply]
Thank you, and if using the UBL firmware, add the three additional lines as mentioned in the readme below G28, but remove the M420 command?

**Reply by CWO4** [reply]
That start looks fine but I would add the "Fade" switch to the M420 command.  M420 S1 Z3   The "Z" fade switch brings the Z axis level back to the actual "real world" mechanical level of the printer.  And takes a load off processing and motor calls.  I fade over 3 layers, hence the "Z3" but you can use any reasonable number you want, the lower the better.  It also increases the accuracy of your print.

**Reply by RB211** [reply]
That start code dragged my extruder into the bed and didn't do the 9 point thing... I've been using a Prusa for the past 5 years and I now realize how sheltered I've been from the real world of 3D printing, this is apparently all new to me.

**Reply by CWO4** [reply]
It crashed your bed because there was apparently no stored mesh in the EPROM.  You have to do a bed level first.  Once a leveling has been done AND STORED, all the M420 line does is recall it.  If the leveling was bad, the recall is bad.  GIGO.

**Comment by RB211** [comment]
I got it to work, I am using UBL, I had to recall the UBL mesh which I had
already done, it is working now that I added those three lines and removed
the other one. You must be referring to the Bi-linear Mesh.

**Comment by RB211** [comment]
Thank you for the help, and a BIG thankyou for authoring this firmware, it is appreciated.
